sw/host/rom_ext_image_tools/signer/image/src/image.rs

```diff
@@ -31,51 +38,89 @@ impl RawImage {
-            manifest::ROM_EXT_IMAGE_TIMESTAMP_OFFSET,
-        );
-        myclosure(
-            &config.extension0_checksum,
-            manifest::ROM_EXT_EXTENSION0_CHECKSUM_OFFSET,
-        );
-        myclosure(
-            &config.extension1_checksum,
-            manifest::ROM_EXT_EXTENSION1_CHECKSUM_OFFSET,
-        );
-        myclosure(
-            &config.extension2_checksum,
-            manifest::ROM_EXT_EXTENSION2_CHECKSUM_OFFSET,
-        );
-        myclosure(
-            &config.extension3_checksum,
-            manifest::ROM_EXT_EXTENSION3_CHECKSUM_OFFSET,
-        );
+
+        let offsets = [
+            (
+                manifest::ROM_EXT_EXTENSION0_OFFSET_OFFSET,
+                manifest::ROM_EXT_EXTENSION0_CHECKSUM_OFFSET,
+            ),
+            (
+                manifest::ROM_EXT_EXTENSION1_OFFSET_OFFSET,
+                manifest::ROM_EXT_EXTENSION1_CHECKSUM_OFFSET,
+            ),
+            (
+                manifest::ROM_EXT_EXTENSION2_OFFSET_OFFSET,
+                manifest::ROM_EXT_EXTENSION2_CHECKSUM_OFFSET,
+            ),
+            (
+                manifest::ROM_EXT_EXTENSION3_OFFSET_OFFSET,
+                manifest::ROM_EXT_EXTENSION3_CHECKSUM_OFFSET,
+            ),
+        ];
+        for (i, offset) in offsets.iter().enumerate() {
+            update(&config.extensions[i].offset, offset.0);
+            update(&config.extensions[i].checksum, offset.1);
+        }
+
+        let usage_constraints_path = &config.input_files.usage_constraints_path;
+        self.update_usage_constraints_field(usage_constraints_path);
+
+        let lockdown_info = &config.peripheral_lockdown_info;
+        self.update_peripheral_lockdown_info_field(lockdown_info);
+
+        // TODO: calculated at runtime, so we should probably rename this
+        //       function.
+        self.update_timestamp_field();
+    }
+
+    /// Updates ROM_EXT manifest signature key public exponent field.
+    pub fn update_exponent_field(&mut self, exponent: &[u8]) {
+        let offset = manifest::ROM_EXT_SIGNATURE_KEY_PUBLIC_EXPONENT_OFFSET;
+        self.update_field(exponent, offset);
+    }
+
+    /// Updates ROM_EXT manifest signature key modulus field.
+    pub fn update_modulus_field(&mut self, modulus: &[u8]) {
+        let offset = manifest::ROM_EXT_SIGNATURE_KEY_MODULUS_OFFSET;
+        self.update_field(modulus, offset);
+    }
+
+    /// Updates ROM_EXT manifest signature field.
+    pub fn update_signature_field(&mut self, signature: &[u8]) {
+        self.update_field(signature, manifest::ROM_EXT_IMAGE_SIGNATURE_OFFSET);
+    }
+
+    /// Returns the portion of the image used for signing.
+    ///
+    /// Manifest identifier and the signature itself are not signed. The rest
+    /// of the image, including all the manifest fields that follow the
+    /// signature field.
+    pub fn data_to_sign(&self) -> &[u8] {
+        let offset = manifest::ROM_EXT_SIGNED_AREA_START_OFFSET as usize;
```

Comment:
Probably same here, but this one actually does something instead of just aliasing the name, so I'm OK with this one here.

Reply:
Keeping, this one :)